### PR TITLE
[#44876] Implement API endpoint for creating upload link

### PR DESCRIPTION
**Pull request by apfohl**
https://community.openproject.org/work_packages/44876

### CODE REVIEW

File: modules/storages/lib/api/v3/storage_files/storage_files_api.rb [hunk 1]
```diff
@@ -54,16 +54,44 @@ def raise_error(error)
...
+            },
+            on_failure: ->(error) { raise_error(error) }
+          )
+      end
+
+      # RequestBody:
+      # {
+      #   "fileName": "ape.png",
+      #   "parent": "/Pictures"
+      # }
```
Comment (Kharonus) [inline modules/storages/lib/api/v3/storage_files/storage_files_api.rb:76]: the request body is only passed through, do we need to have a inline comment here?

File: modules/storages/spec/lib/api/v3/storage_files/storage_upload_link_representer_spec.rb [hunk 2]
```diff
@@ -66,28 +66,5 @@
...
-        end
-      end
-
-      describe 'to finalize' do
-        it_behaves_like 'has an untitled link' do
-          let(:link) { 'finalize' }
-          let(:href) { finalize }
-        end
-      end
-    end
```
Comment (Kharonus) [inline modules/storages/spec/lib/api/v3/storage_files/storage_upload_link_representer_spec.rb:91]: why did you remove the specs for the finalize link?

Comment (apfohl) [reply modules/storages/spec/lib/api/v3/storage_files/storage_upload_link_representer_spec.rb:91]: Because they get implemented in a succeeding WP: https://community.openproject.org/projects/openproject/work_packages/44880/activity

I wanted to keep this clean of thinks that do not matter in this WP here.

File: modules/storages/lib/api/v3/storage_files/storage_files_api.rb (hunk 1 above)
Comment (apfohl) [reply modules/storages/lib/api/v3/storage_files/storage_files_api.rb:76]: No, I will remove it in one of the upcoming WPs.

Comment (apfohl) [reply modules/storages/lib/api/v3/storage_files/storage_files_api.rb:76]: I'd like to keep it now for reference.

**Review by Kharonus — APPROVED** [review]
LGTM
